.github/workflows/build.yml

```diff
@@ -0,0 +1,106 @@
+name: Build and Test ldsCtrlEst
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Cache vcpkg binaries
+      uses: actions/cache@v3
+      with:
+        path: ${{ runner.os == 'Windows' && 'C:\\Users\\runneradmin\\AppData\\Local\\vcpkg\\' || '~/.cache/vcpkg' }}
+        key: vcpkg-binary-${{ matrix.os }}-${{ hashFiles('vcpkg.json') }}
+        restore-keys: |
+          vcpkg-binary-${{ matrix.os }}-
+    - name: Cache vcpkg directory
+      uses: actions/cache@v3
+      with:
+        path: vcpkg
+        key: vcpkg-${{ matrix.os }}-${{ hashFiles('vcpkg.json') }}
+        restore-keys: |
+          vcpkg-${{ matrix.os }}-
+    - name: Set up vcpkg for Windows
+      if: runner.os == 'Windows'
+      shell: pwsh
+      run: |
+        if (Test-Path -Path "vcpkg") {
+          Remove-Item -Recurse -Force "vcpkg"
+        }
+        git clone https://github.com/microsoft/vcpkg.git
+        cd vcpkg
+        .\bootstrap-vcpkg.bat
+        .\vcpkg integrate install
+    - name: Set up vcpkg for macOS/Linux
+      if: runner.os != 'Windows'
+      run: |
+        rm -rf vcpkg
+        git clone https://github.com/microsoft/vcpkg.git
+        cd vcpkg
+        ./bootstrap-vcpkg.sh
+        ./vcpkg integrate install
+    - name: Set up dependencies for macOS and Linux
+      if: runner.os != 'Windows'
+      run: |
+        if [ "${{ runner.os }}" == "macOS" ]; then
+          brew install cmake gfortran curl zip unzip gnu-tar
```

Comment:
can we leave out gfortran and let vcpkg take care of it? we want to minimize these dependency installation steps that don't get cached

Reply:
some of these tools come pre-installed and some have a convenient GitHub action to set up to speed up this step on all three OS's

Reply:
For an example of pre-installed software available on Ubuntu: https://github.com/actions/runner-images/blob/main/images/ubuntu/toolsets/toolset-2404.json

Reply:
It's recommended to use actions when possible. we might want to do this with things like clang and cmake build/configure: 
- https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#using-preinstalled-software
- https://github.com/marketplace/actions/install-llvm-and-clang
- https://github.com/marketplace/actions/run-cmake